src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+        if (index == size) {
+            linkLast(value);
```

Comment:
```suggestion
            add(value);
```
what if you call `new MyLinkedList<Integer>().add(3, 0)`?

Reply:
Of course.. Done.